master/buildbot/process/buildrequestdistributor.py

```diff
@@ -504,12 +504,12 @@ def _maybeStartBuildsOnBuilder(self, bldr, _reactor=reactor):
                 bc = self.createBuildChooser(bldr, self.master)
                 continue
 
-            buildStarted = yield bldr.maybeStartBuild(worker, breqs)
-            if not buildStarted:
-                yield self.master.data.updates.unclaimBuildRequests(brids)
-                # try starting builds again.  If we still have a working worker,
-                # then this may re-claim the same buildrequests
-                self.botmaster.maybeStartBuildsForBuilder(self.name)
+            d = bldr.maybeStartBuild(worker, breqs)
+
+            @d.addCallback
+            def isItStarted(buildStarted):
+                if not buildStarted:
+                    yield self.master.data.updates.unclaimBuildRequests(brids)
```

Comment:
> I'm not sure what you're trying to do with this change -- it looks like you've just taken a correct use of inlineCallbacks and replaced it with an incorrect one (which would lead logically to test failures or worse)

Yeah the code is not anywhere near ready to merge.. I just wanted to get something out there to have a discussion on how this might be done.

Basically, what I would like to do is have Latent workers not block while they're launching so they can launch in parallel. Launching 20-40 latent EC2 workers (2.5-3.5 min each) sequentially bogs CI down considerably.

I got some help from @tardyp on IRC, but I'm not familiar with twisted at all, so it's been difficult for me to hack away at this.